src/libstore/meson.build

```diff
@@ -0,0 +1,452 @@
+project('nix-store', 'cpp',
+  version : files('.version'),
+  default_options : [
+    'cpp_std=c++2a',
+    # TODO(Qyriad): increase the warning level
+    'warning_level=1',
+    'debug=true',
+    'optimization=2',
+    'errorlogs=true', # Please print logs for tests that fail
+  ],
+  meson_version : '>= 1.1',
+  license : 'LGPL-2.1-or-later',
+)
+
+cxx = meson.get_compiler('cpp')
+
+# See note in ../nix-util/meson.build
+deps_private = [ ]
+
+# See note in ../nix-util/meson.build
+deps_public = [ ]
+
+# See note in ../nix-util/meson.build
+deps_other = [ ]
+
+configdata = configuration_data()
+
+# TODO rename, because it will conflict with downstream projects
+configdata.set_quoted('PACKAGE_VERSION', meson.project_version())
+
+configdata.set_quoted('SYSTEM', host_machine.system())
```

Comment:
This is probably wrong. `host_machine.system()` is expected to have e.g. `'linux'` in other places.

I don't know how to get the right value, reliably, and we've [defined it to be based on autoconf](https://nix.dev/manual/nix/2.22/contributing/hacking.html?highlight=system#system-type), which I don't think we want to continue doing, or do we? Alternatively, we could leave it to the packager, and pass `stdenv.hostPlatform.system` in our own packaging?